Thanks for updating the test. It doesn't seem to be passing for me, though:

```
FAIL: test_filters_markdown (hamlpy.test.template_compare_test.TestTemplateCompare)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/nsh/projects/HamlPy/hamlpy/test/template_compare_test.py", line 47, in test_filters_markdown
self._compare_test_files('filtersMarkdown')
  File "/home/nsh/projects/HamlPy/hamlpy/test/template_compare_test.py", line 124, in _compare_test_files
eq_(parsed, html)
AssertionError: u'<p>hello</p>\n<p>no paragraph</p>\n<p>line break  </p>\n<p>follow</p>\n<p>New paragraph</p>\n<pre><code>code block\n</code></pre>' != '<p>hello\nno paragraph</p>\n<p>line break<br />\nfollow</p>\n<p>New paragraph</p>\n<pre><code>code block\n</code></pre>'
-------------------- >> begin captured stdout << ---------------------

HTML (actual): 
1. <p>hello</p>
2. <p>no paragraph</p>
3. <p>line break  </p>
4. <p>follow</p>
5. <p>New paragraph</p>
6. <pre><code>code block
7. </code></pre>
Difference begins at line 1 column 9
HTML (actual, len=12)   : <p>hello</p>
HTML (expected, len= 8) : <p>hello
Character code (actual)  : 60 (<)
Character code (expected): 10 (
)

--------------------- >> end captured stdout << ----------------------
```

--- a/hamlpy/nodes.py
+++ b/hamlpy/nodes.py
@@ -593,10 +593,14 @@ class MarkdownFilterNode(FilterNode):
         if self.children:
             if not _markdown_available:
                 raise NotAvailableError("Markdown is not available")
-
             self.before = self.render_newlines()[1:]
             indent_offset = len(self.children[0].spaces)
-            text = ''.join(''.join([c.spaces[indent_offset:], c.raw_haml.lstrip(), c.render_newlines()]) for c in self.children)
-            self.before += markdown(text)
+            lines = []
+            for c in self.children:
+                haml = c.raw_haml.lstrip()
+                if haml[-1] == '\n':
+                    haml = haml[:-1]
+                lines.append(c.spaces[indent_offset:] + haml + c.render_newlines())
+            self.before += markdown( ''.join(lines))
         else:
             self.after = self.render_newlines()

